posthog/temporal/workflows/cohort_for_flag.py

```diff
@@ -0,0 +1,265 @@
+import dataclasses
+import datetime
+import json
+from typing import Dict
+from django.conf import settings
+from django.db import DatabaseError
+from rest_framework.exceptions import ValidationError
+from asgiref.sync import sync_to_async, async_to_sync
+from temporalio.client import Client
+
+import structlog
+from posthog.models.feature_flag.flag_matching import (
+    FeatureFlagMatcher,
+    FlagsMatcherCache,
+    get_feature_flag_hash_key_overrides,
+)
+from posthog.models.filters.filter import Filter
+from posthog.models.person.person import PersonDistinctId
+from posthog.models.property.property import Property, PropertyGroup
+from posthog.queries.base import property_group_to_Q
+from django.db.models import Prefetch, prefetch_related_objects, OuterRef, Subquery
+from posthog.constants import PropertyOperatorType
+from sentry_sdk import capture_exception
+from posthog.models import Cohort, FeatureFlag, Person
+from temporalio import activity, workflow
+
+from posthog.temporal.workflows.base import PostHogWorkflow
+
+logger = structlog.get_logger(__name__)
+
+
+@dataclasses.dataclass
+class CreateCohortForFlagWorkflowInputs:
+    """Inputs to the cohort_for_flag activity.
+
+    Attributes:
+        team_id: The id of the team
+        cohort_id: The id of the cohort to populate.
+        flag: The feature flag to use for matching persons.
+        batchsize: The number of persons to process at a time.
+    """
+
+    team_id: int
+    cohort_id: int
+    flag: str
+    batchsize: int
+
+
+@async_to_sync
+async def start_cohort_from_flag_workflow(temporal: Client, inputs: CreateCohortForFlagWorkflowInputs) -> str:
+    workflow_id = f"{inputs.team_id}-cohort-{inputs.cohort_id}-for-flag-{inputs.flag}"
+    await temporal.start_workflow(
+        "cohort_for_flag",
+        inputs,
+        id=workflow_id,
+        task_queue=settings.TEMPORAL_TASK_QUEUE,
+    )
```

Comment:
Temporal will raise an exception if you try to start a workflow with the same id as one that has already started. I may be wrong but it seems like the workflow id can be not-unique when repeating API calls. Do we have any rail-guards in place somewhere else to check that this workflow will be triggered only once?

Reply:
Every API call generates a new cohort id, so this should be ok